packages/okta-react/CHANGELOG.md

```diff
@@ -1,3 +1,14 @@
+# 2.0.0
+
+### Features
+
+- Now offers the following React Hooks (2.x requires React 16.8+)
+  - `useAuth` 
```

Comment:
How about only provide one hook, then devs can grab what they need from the hook. Something like:
```
const { state, login, ...authfunctions }  = useAuth();
```

Reply:
The main reason I separated them is that they serve different purposes.  `useAuth` allows you to manipulate the auth state, while `useAuthState` is for READING the auth state.  That means the difference serves a communication purpose - someone reviewing a component that manipulates the auth state has that jump out as obvious.

Reply:
I get your point. But still feel it might be easier just provide one single hook, since state can still be accessed very easily by destructing the return of hook.

Reply:
"easier" in the sense of having one object for two purposes, sure.  But as I pointed out, that comes at a cost of communication, which is the goal of good programming - to communicate.

Reply:
Agreed. As we are rolling out new contract for dev, just want to explore other options a bit.

Reply:
Maybe `useAuthService` would communicate more clearly? (I sometimes get confused referring to `Auth.js` (with a capital 'A') because of ambiguity with the underlying okta-auth-js. We could rename it to `AuthService`, this would match the naming in angular and indicates it provides methods (which can produce side effects. reading authState does not change anything).

Reply:
There's a lot there, let's break it down:

### rename Auth to AuthService

I'm (mostly) fine with this.  The only reason I'm mostly fine is I don't know that the "AuthService" concept is generally understood - I tend to the think of services in terms of web calls and not generic functionality providers, but I don't know if that's just me.

If we do this, it should be consistent throughout:
- Auth.js to AuthService.js
- `auth` param to `<Security>` becomes `authService`
- `useAuth` becomes `useAuthService`
- prop becomes `authService` instead of `auth`
- I'd leave `withOktaAuth` the same and still provide both props (though change `auth` to `authService` unless convinced otherwise (see below)

Does this make sense?  If not, I'd say leave it as-is.    

### The HOC

I have one HOC but two Hooks because the HOC is kind of annoying to use more than one of. ( `annoying(toUseMoreThen(OneOf)) `  

I think there's a communication benefit to having two hooks - using just `authState` says you aren't changing the state, just reading it, while `auth` (or `authService`) declares you are changing it, making it obvious to anyone skimming the code.    (userinfo is a hole here, but we've talked about putting it on the authState in the future)

I couldn't tell from your comment which way your were advocating.  Given my arguments, do you have a stance one way or the other?

Reply:
I like everything about changing "auth" to "authService". We are using "Auth Service" in the README because "Auth Object" doesn't mean much. I understand a service to be a singleton object which exposes public methods. It is something you *use*.

I understand what you are saying about communication, but I think a single effect will be easy to understand and we would future-proof ourselves better with a single effect that passes ALL the objects that we are providing now or in the future. Our docs should show using the array syntax `['auth', 'authState']` on your effect so you only re-render on the props that you care about. I think this would be a good practice vs. rendering whenever "anything" changes. I also like the symmetry between the effect and the HOC. If we add another property, we will add it to both.